algopy_testing/src/algopy/op.py

```diff
@@ -168,3 +178,398 @@ def vrf_verify(
+    result = extract(a, b, 8)
+    result_int = int.from_bytes(result.value)
+    return UInt64(result_int)
+
+
+def getbit(a: Bytes | UInt64 | bytes | int, b: UInt64 | int, /) -> UInt64:
+    if isinstance(a, Bytes | bytes):
+        return _getbit_bytes(a, b)
+    if isinstance(a, UInt64 | int):
+        a_bytes = _uint64_to_bytes(a)
+        return _getbit_bytes(a_bytes, b, "little")
+    raise ValueError("Unknown type for argument a")
+
+
+def getbyte(a: Bytes | bytes, b: UInt64 | int, /) -> UInt64:
+    a = as_bytes(a)
+    int_list = list(a)
+
+    max_index = len(int_list) - 1
+    b = as_int(b, max=max_index)
+
+    return UInt64(int_list[b])
+
+
+def itob(a: UInt64 | int, /) -> Bytes:
+    return Bytes(_uint64_to_bytes(a))
+
+
+def mulw(a: UInt64 | int, b: UInt64 | int, /) -> tuple[UInt64, UInt64]:
+    a = as_int64(a)
+    b = as_int64(b)
+    result = a * b
+    return _int_to_uint128(result)
+
+
+def replace(a: Bytes | bytes, b: UInt64 | int, c: Bytes | bytes, /) -> Bytes:
+    a = a if (isinstance(a, Bytes)) else Bytes(a)
+    b = as_int64(b)
+    c = as_bytes(c)
+    if b + len(c) > len(a):
+        raise ValueError(f"expected value <= {len(a)}, got: {b + len(c)}")
+    return a[slice(0, b)] + c + a[slice(b + len(c), len(a))]
+
+
+def select_bytes(a: Bytes | bytes, b: Bytes | bytes, c: bool | UInt64 | int, /) -> Bytes:
+    a = as_bytes(a)
+    b = as_bytes(b)
+    c = int(c) if (isinstance(c, bool)) else as_int64(c)
+    return Bytes(b if c != 0 else a)
+
+
+def select_uint64(a: UInt64 | int, b: UInt64 | int, c: bool | UInt64 | int, /) -> UInt64:
+    a = as_int64(a)
+    b = as_int64(b)
+    c = int(c) if (isinstance(c, bool)) else as_int64(c)
+    return UInt64(b if c != 0 else a)
+
+
+def setbit_bytes(a: Bytes | bytes, b: UInt64 | int, c: UInt64 | int, /) -> Bytes:
+    return _setbit_bytes(a, b, c)
+
+
+def setbit_uint64(a: UInt64 | int, b: UInt64 | int, c: UInt64 | int, /) -> UInt64:
+    a_bytes = _uint64_to_bytes(a)
+    result = _setbit_bytes(a_bytes, b, c, "little")
+    return UInt64(int.from_bytes(result.value))
+
+
+def setbyte(a: Bytes | bytes, b: UInt64 | int, c: UInt64 | int, /) -> Bytes:
+    a = as_bytes(a)
+    int_list = list(a)
+
+    max_index = len(int_list) - 1
+    b = as_int(b, max=max_index)
+    c = as_int8(c)
+
+    int_list[b] = c
+    return Bytes(_int_list_to_bytes(int_list))
+
+
+def shl(a: UInt64 | int, b: UInt64 | int, /) -> UInt64:
+    a = as_int64(a)
+    b = as_int(b, max=63)
+    result = (a * (2**b)) % (2**64)
+    return UInt64(result)
+
+
+def shr(a: UInt64 | int, b: UInt64 | int, /) -> UInt64:
+    a = as_int64(a)
+    b = as_int(b, max=63)
+    result = a // (2**b)
+    return UInt64(result)
+
+
+def sqrt(a: UInt64 | int, /) -> UInt64:
+    a = as_int64(a)
+    return UInt64(math.isqrt(a))
+
+
+def substring(a: Bytes | bytes, b: UInt64 | int, c: UInt64 | int, /) -> Bytes:
+    a = a if (isinstance(a, Bytes)) else Bytes(a)
```

Comment:
minor: using as_bytes here and then converting to `Bytes` at the end might be easier to read?

Reply:
It is fixed in the commit 424cb6c20b396b310e5a68a83966f03e28642d89.